### PR TITLE
Aaron/is pending kill to is valid

### PR DESCRIPTION
**Pull request by Blyron**
Change IsPendingToKill() -> IsValid()

IsPendingKill has been deprecated. We have to use IsValid from now on

### CODE REVIEW

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Actor/ActorBlueprintFunctionLibrary.cpp [hunk 1]
```diff
@@ -1523,14 +1523,14 @@ FColor UActorBlueprintFunctionLibrary::RetrieveActorAttributeToColor(
 // editor crashing while people are testing new actor definitions.
 #if WITH_EDITOR
 #  define CARLA_ABFL_CHECK_ACTOR(ActorPtr)                    \
-  if ((ActorPtr == nullptr) || ActorPtr->IsPendingKill())     \
+  if ((ActorPtr == nullptr) || IsValid(ActorPtr))     \
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Actor/ActorBlueprintFunctionLibrary.cpp:1526]:
```suggestion
  if ((ActorPtr == nullptr) || !IsValid(ActorPtr))     \
```
I think you wanted to look for non valid Actors to throw an error right?

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Actor/ActorBlueprintFunctionLibrary.cpp [hunk 2]
```diff
@@ -1523,14 +1523,14 @@ FColor UActorBlueprintFunctionLibrary::RetrieveActorAttributeToColor(
...
   {                                                           \
     UE_LOG(LogCarla, Error, TEXT("Cannot set empty actor!")); \
     return;                                                   \
   }
 #else
 #  define CARLA_ABFL_CHECK_ACTOR(ActorPtr) \
-  check((ActorPtr != nullptr) && !ActorPtr->IsPendingKill());
+  check((ActorPtr != nullptr) && IsValid(ActorPtr) );
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Actor/ActorBlueprintFunctionLibrary.cpp:1533]:
```suggestion
  check((ActorPtr != nullptr) && !IsValid(ActorPtr));
```
I think you wanted to look for non valid Actors to throw an error right?

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Server/CarlaServer.cpp [hunk 3]
```diff
@@ -2149,7 +2149,7 @@ BIND_SYNC(is_sensor_enabled_for_ros) << [this](carla::streaming::detail::stream_
         else
         {
           auto Actor = View.GetActor();
-          if (!Actor->IsPendingKill())
+          if ( IsValid(Actor) )
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Server/CarlaServer.cpp:2152]:
```suggestion
          if (IsValid(Actor))
```

Formatting.

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp [hunk 4]
```diff
@@ -48,7 +48,7 @@ void UCarlaSettingsDelegate::RegisterSpawnHandler(UWorld *InWorld)
 void UCarlaSettingsDelegate::OnActorSpawned(AActor *InActor)
 {
   check(CarlaSettings != nullptr);
-  if (InActor != nullptr && IsValid(InActor) && !InActor->IsPendingKill() &&
+  if (InActor != nullptr && IsValid(InActor) &&
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp:51]:
```suggestion
  if (IsValid(InActor) &&
```

IsValid() already checks if InActor is null.

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp [hunk 5]
```diff
@@ -126,7 +126,7 @@ void UCarlaSettingsDelegate::ApplyQualityLevelPreRestart()
 {
   CheckCarlaSettings(nullptr);
   UWorld *InWorld = CarlaSettings->GetWorld();
-  if (!IsValid(InWorld) || InWorld->IsPendingKill())
+  if ( !IsValid(InWorld) )
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp:129]:
```suggestion
  if (!IsValid(InWorld))
```
Formatting.

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp [hunk 6]
```diff
@@ -158,7 +158,7 @@ void UCarlaSettingsDelegate::CheckCarlaSettings(UWorld *world)
   {
     return;
   }
-  if (world == nullptr || !IsValid(world) || world->IsPendingKill())
+  if (world == nullptr || !IsValid(world) )
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp:161]:
```suggestion
  if (IsValid(world))
```
IsValid() already checks if world is null

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp [hunk 7]
```diff
@@ -232,12 +232,12 @@ void UCarlaSettingsDelegate::SetAllRoads(
     const float max_draw_distance,
     const TArray<FStaticMaterial> &road_pieces_materials) const
 {
-  if (!world || !IsValid(world) || world->IsPendingKill())
+  if (!world || !IsValid(world) )
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp:235]:
```suggestion
  if (!world || !IsValid(world))
```
Formatting.

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp [hunk 8]
```diff
@@ -232,12 +232,12 @@ void UCarlaSettingsDelegate::SetAllRoads(
...
   {
     return;
   }
   AsyncTask(ENamedThreads::GameThread, [=]() {
-    if (!world || !IsValid(world) || world->IsPendingKill())
+    if (!world || !IsValid(world) )
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp:240]:
```suggestion
    if (!world || !IsValid(world))
```
Formatting.

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp [hunk 9]
```diff
@@ -247,7 +247,7 @@ void UCarlaSettingsDelegate::SetAllRoads(
     for (int32 i = 0; i < actors.Num(); i++)
     {
       AActor *actor = actors[i];
-      if (!IsValid(actor) || actor->IsPendingKill())
+      if (!IsValid(actor) )
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp:250]:
```suggestion
      if (!IsValid(actor))
```
Formatting.

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Actor/ActorBlueprintFunctionLibrary.cpp (hunk 1 above)
Comment (Blyron) [reply Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Actor/ActorBlueprintFunctionLibrary.cpp:1526]: You are right

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Actor/ActorBlueprintFunctionLibrary.cpp (hunk 2 above)
Comment (Blyron) [reply Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Actor/ActorBlueprintFunctionLibrary.cpp:1533]: No. the check will trigger if one of conditions are false. We want to check if the ActorPtr is not null ptr and if it is valid

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Actor/ActorBlueprintFunctionLibrary.cpp [hunk 10]
```diff
@@ -1523,14 +1523,14 @@ FColor UActorBlueprintFunctionLibrary::RetrieveActorAttributeToColor(
...
-  if ((ActorPtr == nullptr) || ActorPtr->IsPendingKill())     \
+  if (!IsValid(ActorPtr))     \
   {                                                           \
     UE_LOG(LogCarla, Error, TEXT("Cannot set empty actor!")); \
     return;                                                   \
   }
 #else
 #  define CARLA_ABFL_CHECK_ACTOR(ActorPtr) \
-  check((ActorPtr != nullptr) && !ActorPtr->IsPendingKill());
+  check((ActorPtr != nullptr) && IsValid(ActorPtr));
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Actor/ActorBlueprintFunctionLibrary.cpp:1533]:
```suggestion
  IsValid(ActorPtr));
```

IsValid is already performing the null check

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp [hunk 11]
```diff
@@ -232,12 +232,12 @@ void UCarlaSettingsDelegate::SetAllRoads(
     const float max_draw_distance,
     const TArray<FStaticMaterial> &road_pieces_materials) const
 {
-  if (!world || !IsValid(world) || world->IsPendingKill())
+  if (!world || !IsValid(world))
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp:235]:
```suggestion
  if (!IsValid(world))
```
IsValid() is already doing the nullcheck.

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp [hunk 12]
```diff
@@ -232,12 +232,12 @@ void UCarlaSettingsDelegate::SetAllRoads(
...
   {
     return;
   }
   AsyncTask(ENamedThreads::GameThread, [=]() {
-    if (!world || !IsValid(world) || world->IsPendingKill())
+    if (!world || !IsValid(world))
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp:240]:
```suggestion
    if (!IsValid(world))
```
IsValid() is already doing the nullcheck.

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp [hunk 13]
```diff
@@ -422,20 +422,20 @@ void UCarlaSettingsDelegate::SetAllLights(
...
     const bool hide_non_directional) const
 {
-  if (!world || !IsValid(world) || world->IsPendingKill())
+  if (!world || !IsValid(world))
   {
     return;
   }
   AsyncTask(ENamedThreads::GameThread, [=]() {
-    if (!world || !IsValid(world) || world->IsPendingKill())
+    if (!world || !IsValid(world))
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp:430]:
```suggestion
    if (!IsValid(world))
```
IsValid() is already doing the nullcheck.

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp [hunk 14]
```diff
@@ -422,20 +422,20 @@ void UCarlaSettingsDelegate::SetAllLights(
     const bool cast_shadows,
     const bool hide_non_directional) const
 {
-  if (!world || !IsValid(world) || world->IsPendingKill())
+  if (!world || !IsValid(world))
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Settings/CarlaSettingsDelegate.cpp:425]:
```suggestion
  if (!IsValid(world))
```
IsValid() is already doing the nullcheck.

**Review by xavisolesoft — CHANGES_REQUESTED** [review]

**Review by xavisolesoft — APPROVED** [review]

**Pull request merged**